aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -50,7 +50,9 @@ struct dists {
 
   // Special case backward when p is less than two
   struct lt_two {
-    static __forceinline__ __device__ scalar_t backward(const scalar_t diff, const scalar_t grad, const scalar_t dist, const scalar_t p) { return dist == 0.0 ? 0 : sign(diff) * std::pow(std::abs(diff), p - 1) * grad / std::pow(dist, p - 1); }
+    static __forceinline__ __device__ scalar_t backward(const scalar_t diff, const scalar_t grad, const scalar_t dist, const scalar_t p) {
+      return (dist == 0.0 || (diff == 0.0 && p < 1)) ? 0 : (sign(diff) * std::pow(std::abs(diff), p - 1) * grad / std::pow(dist, p - 1));
```

Comment:
```suggestion
      return (dist == 0.0 || (diff == 0.0 && p < 1)) ? 0 : (sign(diff) * std::pow(std::abs(diff), p - 1) * grad * std::pow(dist, 1 - p));
```
This resolves division by zero when `cdist` is close to zero, so should be more numerically stable.

Reply:
Or probably even more numerically stable approach:
`std::pow(std::abs(diff) / dist, p - 1) * grad`, or maybe we should try log-space, although it seems bad for zero.
`

Reply:
That change does not avoid the NaN by itself. It still needs the `(diff == 0.0 && p < 1)` check. Because of that, I feel like changing the formula in this way would be outside the scope of this PR, right?

Reply:
It will be bc-breaking for sure, so I agree with you, better be kept for a different PR specifically targeting overall numerical stability.